test/e2e/vsphere_volume_placement.go

```diff
@@ -62,319 +80,268 @@ var _ = framework.KubeDescribe("Volume Placement [Feature:Volume]", func() {
+			verifyFilesExistOnVSphereVolume(ns, podB.Name, podBFiles)
+
+			By("Deleting pod-A")
+			err = c.CoreV1().Pods(ns).Delete(podA.Name, nil)
 			Expect(err).NotTo(HaveOccurred())
-			volumePaths = append(volumePaths, volumePath)
-			testvolumePathsPodB = append(testvolumePathsPodA, volumePath)
-
-			for index := 0; index < 5; index++ {
-				By("Creating pod-A on the node: " + node1Name + " with volume:" + testvolumePathsPodA[0])
-				podA = createPodWithVolumeAndNodeSelector(c, ns, vsp, node1Name, node1KeyValueLabel, testvolumePathsPodA)
-				By("Creating pod-B on the node: " + node1Name + " with volume:" + testvolumePathsPodB[0])
-				podB = createPodWithVolumeAndNodeSelector(c, ns, vsp, node1Name, node1KeyValueLabel, testvolumePathsPodB)
-
-				podAFileName := "/mnt/volume1/" + "podA_" + ns + "-" + strconv.Itoa(index+1) + ".txt"
-				podBFileName := "/mnt/volume1/" + "podB_" + ns + "_" + strconv.Itoa(index+1) + ".txt"
-
-				podAFiles = append(podAFiles, podAFileName)
-				podBFiles = append(podBFiles, podBFileName)
-
-				// Create empty files on the mounted volumes on the pod to verify volume is writable
-				By("Creating empty file on volume mounted on pod-A")
-				createEmptyFileOnVSphereVolume(ns, podA.Name, podAFileName)
-
-				By("Creating empty file volume mounted on pod-B")
-				createEmptyFileOnVSphereVolume(ns, podB.Name, podBFileName)
-
-				// Verify newly and previously created files present on the volume mounted on the pod
-				By("Verify newly Created file and previously created files present on volume mounted on pod-A")
-				verifyFilesExistOnVSphereVolume(ns, podA.Name, podAFiles)
-				By("Verify newly Created file and previously created files present on volume mounted on pod-B")
-				verifyFilesExistOnVSphereVolume(ns, podB.Name, podBFiles)
-
-				By("Deleting pod-A")
-				err = c.CoreV1().Pods(ns).Delete(podA.Name, nil)
-				Expect(err).NotTo(HaveOccurred())
-				podA = nil
-
-				By("Deleting pod-B")
-				err = c.CoreV1().Pods(ns).Delete(podB.Name, nil)
-				Expect(err).NotTo(HaveOccurred())
-				podB = nil
-			}
-		})
+
+			By("Deleting pod-B")
+			err = c.CoreV1().Pods(ns).Delete(podB.Name, nil)
+			Expect(err).NotTo(HaveOccurred())
+		}
 	})
 })
 
-func testSetupVolumePlacement(client clientset.Interface, namespace string, nodes *v1.NodeList) (node1Name string, node1KeyValueLabel map[string]string, node2Name string, node2KeyValueLabel map[string]string) {
-	if len(nodes.Items) != 0 {
-		node1Name = nodes.Items[0].Name
-		node2Name = nodes.Items[1].Name
-	} else {
-		framework.Failf("Unable to find ready and schedulable Node")
+func testSetupVolumePlacement(client clientset.Interface, namespace string) (node1Name string, node1KeyValueLabel map[string]string, node2Name string, node2KeyValueLabel map[string]string) {
+	nodes := framework.GetReadySchedulableNodesOrDie(client)
+	if len(nodes.Items) < 2 {
+		framework.Skipf("Requires at least %d nodes (not %d)", 2, len(nodes.Items))
```

Comment:
Ok. _Should_ you set `TestContext.CloudConfig.NumNodes` when provider is vsphere?